magento/magento2
# Fixed issue #27039: Sales emails asynchronous sending sends out old emails

### Description (*)
Fixed filter for async email sending and refactoring the class as well.


### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes magento/magento2#27039

### Manual testing scenarios (*)

1. Create new Orders
2. Enable the [email async sending](https://docs.magento.com/user-guide/configuration/sales/sales-emails.html)
3. Check if it only sends emails for Orders created one day before the new configuration has been enabled.

### Questions or comments

The bugfix itself is only the change in the condition for **created_at** in the collection filter.

From:
`$collection->addFieldToFilter('created_at', ['from' => $startFromDate]);`

To:
`$collection->addFieldToFilter('created_at', ['gteq' => $startFromDate]);`

All others changes is related to the refactoring of the class.

### Contribution checklist (*)
